{{cookiecutter.project_name}}/{{cookiecutter.app_name}}/commons/apispec.py

```diff
@@ -46,11 +46,14 @@ def init_app(self, app):
         app.config.setdefault("SWAGGER_UI_URL", "/swagger-ui")
         app.config.setdefault("SWAGGER_URL_PREFIX", None)
 
+        default_security = [{"jwt": []}]
```

Comment:
Since we're in the extension code here, what do you think of passing `**kwargs` to `init_app` (and `__init__` by extension) instead of setting it directly in the code ? That will require an extra setting to pass on app factory but will allow users to update it directly in app factory instead of updating extension code if they decide to switch from jwt to oauth for example.

Then we can pass our `**kwargs` to `APISpec`, this will also allow users to update more settings if required

Reply:
Sure! I can do that. I can also add `security_type` as an option to #24 and allow the user to choose.

Reply:
Letting users choose their `security_type` may conflict with how authentication is implemented actually, but since we declare security scheme in app factory, it makes more sense to only have to update it in one place.

For `**kwargs` we could also forward them all to `APISpec`:

```python
    def __init__(self, app=None, **kwargs):
        self.spec = None
        if app is not None:
            self.init_app(app, **kwargs)

    def init_app(self, app, **kwargs):
        ...
        self.spec = APISpec(
            title=app.config["APISPEC_TITLE"],
            version=app.config["APISPEC_VERSION"],
            openapi_version=app.config["OPENAPI_VERSION"],
            plugins=[MarshmallowPlugin(), FlaskRestfulPlugin()],
            **kwargs
        )
```

So it will allow us to pass any other top-level key at app factory level (https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#openapi-object) and (https://apispec.readthedocs.io/en/latest/api_core.html#apispec.APISpec)

And then in `app.py`

```python
def configure_apispec(app):
    """Configure APISpec for swagger support
    """
    apispec.init_app(app, security=[{"jwt": []}])
    apispec.spec.components.security_scheme("jwt", {
        "type": "http",
        "scheme": "bearer",
        "bearerFormat": "JWT",
    })
    apispec.spec.components.schema(
        "PaginatedResult", {
            "properties": {
                "total": {"type": "integer"},
                "pages": {"type": "integer"},
                "next": {"type": "string"},
                "prev": {"type": "string"},
            }})

```

Reply:
Got it. I'll push up a change soon.